employees/views.py

```diff
@@ -37,41 +39,80 @@ def query_as_dict(query_set):
+                members__id=self.request.user.id
+            ).order_by('name')
+        return reports_serializer
+
+    def initial(self, request, *args, **kwargs):
+        super().initial(request, *args, **kwargs)
+        self.reports_dict = query_as_dict(self.get_queryset())
+        self.project_form = ProjectJoinForm(
+            queryset=Project.objects.exclude(members__id=self.request.user.id).order_by('name'),
+        )
+
     def get(self, request):
-        reports_serializer = ReportSerializer(context={'request': request})
-        reports_dict = query_as_dict(self.get_queryset())
         return Response({
-            'serializer': reports_serializer,
-            'reports_dict': reports_dict,
+            'serializer': self._create_serializer(),
+            'reports_dict': self.reports_dict,
             'UI_text': ReportListStrings,
+            'project_form': self.project_form,
         })
 
     def post(self, request):
         reports_serializer = ReportSerializer(data=request.data, context={'request': request})
-        reports_dict = query_as_dict(self.get_queryset())
+
+        if 'join' in request.POST:
+            project_id = request.POST['projects']
+            project = Project.objects.get(id=int(project_id))
+            self._add_project(serializer=reports_serializer, project=project)
+            self.project_form = ProjectJoinForm(
+                queryset=Project.objects.exclude(members__id=self.request.user.id).order_by('name'),
+            )
+            reports_serializer = self._create_serializer()
+            reports_serializer.fields['project'].initial = project
+            return Response({
+                'serializer': reports_serializer,
+                'reports_dict': self.reports_dict,
+                'UI_text': ReportListStrings,
+                'project_form': self.project_form,
+            })
+
         if not reports_serializer.is_valid():
             return Response({
                 'serializer': reports_serializer,
-                'reports_dict': reports_dict,
+                'reports_dict': self.reports_dict,
                 'errors': reports_serializer.errors,
                 'UI_text': ReportListStrings,
+                'project_form': self.project_form,
             })
         reports_serializer.save(author=self.request.user)
-        reports_dict = query_as_dict(self.get_queryset())
-        reports_serializer = ReportSerializer(context={'request': request})
+        self.reports_dict = query_as_dict(self.get_queryset())
         return Response({
-            'serializer': reports_serializer,
-            'reports_dict': reports_dict,
+            'serializer': self._create_serializer(),
+            'reports_dict': self.reports_dict,
             'UI_text': ReportListStrings,
+            'project_form': self.project_form,
         }, status=201)
-        # QUESTION: Should I use some sort of database verification prior to returning Response with status 201?
```

Comment:
This should be deleted in previous PR. What `save()` method returns if something goes wrong should be investigated.